src/components/CardDemo/card-demo.jsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import CodeMirror from 'react-codemirror';
+import 'codemirror/mode/javascript/javascript';
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/addon/lint/lint.css';
+
+import Button from 'terra-button';
+import Text from 'terra-text';
+import ErrorView from 'terra-clinical-error-view';
+
+import { storeCardDemoJson } from '../../actions/card-demo-actions';
+import styles from './card-demo.css';
+import { Card } from '../Card/card';
+
+export class CardDemo extends Component {
+  constructor(props) {
+    super(props);
+    try {
+      const checkJson = JSON.parse(this.props.tempUserJson);
+      if (checkJson && typeof checkJson === 'object') {
+        this.state = {
+          displayJSONError: false,
+          errorText: '',
+        };
+      }
+    } catch (e) {
+      this.state = {
+        displayJSONError: true,
+        errorText: 'Cannot parse the JSON',
+      };
+    }
+    this.state = {
+      displayJSONError: false,
+      errorText: '',
+    };
+
+    this.getExampleCard = this.getExampleCard.bind(this);
+    this.updateCard = this.updateCard.bind(this);
+    this.resetExample = this.resetExample.bind(this);
+  }
+
+  componentDidMount() {
+    const exampleCard = JSON.stringify(this.getExampleCard(), null, 2);
+    if (!this.props.tempUserJson) {
+      this.updateCard(exampleCard);
+    }
+  }
+
+  getExampleCard() {
+    return {
+      summary: 'Example Card',
+      indicator: 'info',
+      detail: 'Add an XYZ complimentary medication OR switch patient order to ABC. ' +
+                'See SMART app for more details.',
+      source: {
+        label: 'Medicine Library',
+        url: 'https://example.com',
+      },
+      links: [
+        {
+          label: 'Medication SMART app',
+          url: 'https://example.com/launch',
+          type: 'smart',
+        },
+      ],
+      suggestions: [
+        {
+          label: 'Add Complimentary',
+          uuid: '123',
+          actions: [
+            {
+              type: 'create',
+              description: 'Add XYZ',
+              resource: {},
+            },
+            {
+              type: 'delete',
+              description: 'Cancel ABC',
+              resource: 'MedicationRequest/ABC',
+            },
+          ],
+        },
+        {
+          label: 'Change Order',
+          uuid: '456',
+          actions: [
+            {
+              type: 'modify',
+              description: 'Modify dosage of Medication',
+              resource: 'MedicationRequest/ABC',
+            },
+          ],
+        },
+      ],
+    };
+  }
+
+  updateCard(newJSON) {
+    try {
+      const parsedJSON = JSON.parse(newJSON);
+      if (parsedJSON && typeof parsedJSON === 'object') {
+        if (this.state.displayJSONError) {
+          this.setState({ displayJSONError: false });
```

Comment:
Should we clear the `errorText` while setting the state for `displayJSONError` to false ?